src/test/shell/integration/cpp_test.sh

```diff
@@ -90,4 +90,30 @@ EOF
   bazel build //a || fail "build failled"
 }
 
+function test_no_recompile_on_shutdown() {
+  mkdir -p a
+  cat > a/BUILD <<EOF
+cc_binary(name="a", srcs=["a.cc"], deps=["b"])
+cc_inc_library(name="b", hdrs=["b.h"])
```

Comment:
Yes, any instance of CppCompileAction will reproduce without inputsKnown starting out as true in the discovery case.  cc_binary, cc_library, cc_test are all affected.